tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -462,6 +468,12 @@ def update_test_metadata_with_transitive_metadata(
             lib_dict['language'] = 'c++'
 
 
+def _get_actual_rule(bazel_rules, rule):
+    while rule != None and rule['actual'] != None:
```

Comment:
`!= None` is bad python afaik. `is not None` is better.

Actually `while rule and rule['actual']:` should be sufficient.